tools/buildmgr/cbuildgen/src/BuildSystemGenerator.cpp

```diff
@@ -98,6 +98,10 @@ bool BuildSystemGenerator::Collect(const string& inputFile, const CbuildModel *m
   m_asMscGlobal = GetString(model->GetTargetAsFlags());
   m_linkerMscGlobal = GetString(model->GetTargetLdFlags());
 
+  m_optimize = model->GetTargetOptimize();
+  m_debug = model->GetTargetDebug();
+  m_warnings = model->GetTargetWarnings();
+
```

Comment:
The options need to be handled also at component/group/file level in addition to target level.

Reply:
Well spotted, but before proceed I prefer to wait for:
- concluding on https://github.com/Open-CMSIS-Pack/devtools/pull/360 and https://github.com/Open-CMSIS-Pack/devtools/issues/179
- and I need to understand how `*flags` are managed at component/group/file level in [cbuildgen]